packages/core/src/baseclient.ts

```diff
@@ -253,13 +253,19 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
       }
     }
 
+    const initialSessionStatus = session.status;
+    const sessionErrorCount = session.errors;
+    const terminalStates = [SessionStatus.Crashed, SessionStatus.Abnormal];
+
     session.update({
       ...(crashed && { status: SessionStatus.Crashed }),
       user,
       userAgent,
-      errors: session.errors + Number(errored || crashed),
+      errors: !sessionErrorCount ? sessionErrorCount + Number(errored || crashed) : sessionErrorCount,
     });
-    this.captureSession(session);
+
+    // Only send a session update if session was not already in a terminal
+    if (!terminalStates.includes(initialSessionStatus)) this.captureSession(session);
```

Comment:
If we include `SessionStatus.Exited` as a terminal state, we can simplify the logic.

```suggestion
    if (initialSessionStatus === SessionStatus.Ok) this.captureSession(session);
```

Reply:
I am leaning more towards keeping the explicit list definition + I do not see any valid usecase where an `captureException` would be called after a session has an Exited status

Reply:
+1 for `initialSessionStatus === SessionStatus.Ok`